openBCIBoard.js

```diff
@@ -142,21 +141,80 @@ function OpenBCIFactory() {
      * @returns {Promise}
      * Author: AJ Keller (@pushtheworldllc)
      */
-    OpenBCIBoard.prototype.boardDisconnect = function() {
+    OpenBCIBoard.prototype.disconnect = function() {
         var self = this;
         var closingPromise = new Promise(function(resolve, reject) {
             if(self.connected === false) { reject(Error('No open serial connection')); }
             self.serial.close(function() {
-                Promise.resolve('Closed the serial connection!');
-            })
+                resolve('Closed the serial connection!');
+            });
+        });
+        return closingPromise;
+    };
+
+    /**
+     * Purpose: To start simulating an open bci board
+     * Note: Must be called after the constructor
+     * @returns {Promise}
+     * Author: AJ Keller (@pushtheworldllc)
+     */
+    OpenBCIBoard.prototype.simulatorStart = function() {
+        var self = this;
+
+        return new Promise(function(resolve,reject) {
+            if(self.isSimulating) {
+                // already running simulator
+                reject('Simulator Already Running');
+            } else {
+                // start simulating
+                self.isSimulating = true;
+                // generateSample is a func that takes the previous sample number
+                var generateSample = OpenBCISample.randomSample(self.numberOfChannels(),self.sampleRate());
+
+                var oldSample = OpenBCISample.newSample();
+                oldSample.sampleNumber = 0;
+                self.simulator = setInterval(function() {
+                    //console.log('Interval...');
+                    var newSample = generateSample(oldSample.sampleNumber);
+                    //console.log(JSON.stringify(newSample));
+                    self.emit('sample',newSample);
+                    oldSample = newSample;
+                    resolve();
+                }, 100);
+
+            }
+
+        });
+    };
+
+    /**
+     * Purpose: To stop simulating an open bci board
+     * Note: Must be called after the constructor
+     * @returns {Promise}
+     * Author: AJ Keller (@pushtheworldllc)
+     */
+    OpenBCIBoard.prototype.simulateStop = function() {
```

Comment:
It should be simulatorStop

--------------------------------------------
 
AJ Keller
Founder & CEO
Push The World LLC
Work: (843) 471-0039
Cell: (203) 733-4538

Warning: The information transmitted is intended only for the person or entity to which it is addressed and may contain confidential and/or privileged material. Any review, retransmission, dissemination or other use of, or taking of any action in reliance upon, this information by persons or entities other than the intended recipient is prohibited. If you received this in error, please contact the sender and destroy this material.
 

> On Dec 31, 2015, at 2:08 PM, Teon L Brooks <notifications@github.com> wrote:
> 
> simulatorStop